### DISCUSSION TITLE
darkmode bgcolor

### DISCUSSION THREAD

**Discussion by HuberTRoy** [vuepress-theme-hope/vuepress-theme-hope#3607] [Q&A] [answered]
### Checklist

- [X] I am using **LTS** version of Node.js.
- [X] I am using the **latest v2 version** of VuePress and VuePress official plugins.
- [X] I am using the **latest v2 version** of vuepress-theme-hope and all plugins in this repo.
- [X] I followed the docs and I double checked my configuration.
- [X] I have searched [FAQ](https://theme-hope.vuejs.press/faq/) and I did not find the answer.

### Package name

vuepress-theme-hope

### Which operating system are you using?

- [X] macOS
- [X] Windows
- [X] Linux

### Describe the bug

https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/4df6e344f7c3651f46b9097e47fbc096a53244dd/packages/theme/templates/index.build.html#L15

The template on prod hardcode the background. 

The configuration background on `palette.scss` not working fully.

**Comment by Mister-Hope** [comment]
Have you ever test that? Normally, body is on html and sharing the same size, so the bgcolor will be applied correctly.

**Comment by Mister-Hope** [comment]
![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/33315834/09d6f837-20ee-4f3e-bb77-e63eea8b824c)
Try not waste others time
![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/33315834/a074cda4-82f0-4526-a018-f4f929d7ae9c)

**Comment by HuberTRoy** [comment]
@Mister-Hope Sorry for waste your time. 

Please see https://github.com/subquery/documentation, check out on 0f9042f4f5a1941df508814a09281e0aa2cd90bc.

Build and serve it. 

The config is darkmode only and have configaration with 
```
$bg-color: (
  light: #060f32,
  dark: #060f32,
);
```

![微信截图_20231107200101](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/10172415/a1ca8989-4121-4154-8cfc-e47507c37da1)

![微信截图_20231107200110](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/10172415/4df931dd-ff55-4549-a5cb-c34b351804d8)

I don't find relative changes on the new version change log. If it has been change.

**Comment by HuberTRoy** [comment]
The key I think is only compile `palette.scss` as `:root` and the hardcode is 
```
      html[data-theme="dark"] {
        --bg-color: #1d1e1f;
      }

      html,
      body {
        background: var(--bg-color);
      }
```
The priority of `html[data-theme="dark"]` greater than `:root`, so `var(--bg-color)` is hardcode not custom.

**Comment by Mister-Hope** [comment]
Hi like I said above, the final color displayed should be body’s bgcolor

**Comment by Mister-Hope** [comment]
I am not sure,  as you are already applying a patch. 

Just take our theme docs as an example:

![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/33315834/242d50bf-d4c8-4480-af5d-5ab01e2ab081)

The body background color comes from css file:

![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/33315834/73a2a065-ed2f-4523-9a36-31541d9be905)

While the color defined in template is already overrided:

![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/33315834/23026bd0-e611-4e43-a017-b725a6a138e9)

Create a new template and try modifying `$bg-color`, everthing should work as expected.

I am curious why you need an `!important`

**Reply by Mister-Hope** [reply]
I converted this into a discussion, and willing to help you on this.

Have you checked your customization part?

**Reply by Mister-Hope** [reply]
Hi please try to upgrade to latest version first, as I actually not sure whether there are bugs in previous versions.

**Comment by Mister-Hope** [comment]
BTW, your customization on my theme looks nice, and it would be great if you can add send a pr to https://theme-hope.vuejs.press/demo/projects.html#docs-using-vuepress-theme-hope

**Reply by HuberTRoy** [reply]
@Mister-Hope The style that have !important is my customize style not compiled with theme scss.  Maybe have fixed by latest version. I have try to upgrade= =, it seems have breaking-change. So I give up= =. Thanks your help.

**Reply by Mister-Hope** [reply]
I send a pr for you, accept that and you can send a pr adding your project to demo list.

**Reply by HuberTRoy** [reply]
@Mister-Hope Thank you~, I will try after other works done.

**Reply by HuberTRoy** [reply]
@Mister-Hope Hi, latest version don't fix this issue. I'm comment all style of customize. Do I build commend wrong? The latest version also would not compile to `html[data-theme=dark]`.

**Reply by Mister-Hope** [reply]
It's working on my machine, devserver I mean

**Reply by Mister-Hope** [reply]
Are you sure you install with frozenlock file

**Reply by HuberTRoy** [reply]
@Mister-Hope Yeah. I clone your commit & nvm install 20 & pnpm install & pnpm docs:build.

**Reply by Mister-Hope** [reply]
I reproduce it, will fix it in the next version very soon.

This is because you disable lightmode, and the issue only appears in prod.

Thanks for reporting this

**Reply by HuberTRoy** [reply]
@Mister-Hope yeah, enable lightmode also have this issue, switch to dark mode will see it.

**Reply by jamesbayly** [reply]
https://github.com/vuepress-theme-hope/vuepress-theme-hope/pull/3608/files

**Comment by Mister-Hope** [comment, marked as answer]
https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/b3db538abf9768744649a265a947a076ca4a15ca

**Reply by Mister-Hope** [reply]
BTW: The template style is actually used to avoid people seeing an opposite background color for the first few frames while loading

**Reply by HuberTRoy** [reply]
Got it. Thanks.

**Comment by HuberTRoy** [comment]
@Mister-Hope 

Hi, there also have a problem. 

The default style is 
```
html[data-theme="dark"] {
    background: #1d1e1f;
}
```

When I browse in Mac, I can over scroll the page and the html's background will show. 
Compiled style cannot override the default style.

```
html, body {
    background: var(--bg-color);
    transition: background var(--color-transition);
}
```

html < html[data-theme="dark"].

**Reply by Mister-Hope** [reply]
Thanks for reporting, https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/972f3a51a1ae61a8eeb7cfbbf8fb26a49007c661 should fix this
